### ISSUE TITLE
chef/centos-7.1 on atlas

### ISSUE THREAD

**Issue by b-dean** [chef/bento#346]
This is a question as much as it is an issue. I was wondering how often [Chef's boxes on Vagrant Cloud / Hashicorp Altas](https://atlas.hashicorp.com/chef) get updated. I'm particularly interested in `chef/centos-7-1`

Also using things like `chef/centos-6.4` is the only reasonable way to get older platforms to work with kitchen-test thanks to [this change](https://github.com/test-kitchen/kitchen-vagrant/commit/a2ef401fb5a644c96165ce836ff3e6379637a37f) made it so all our tests with multiple older platforms in the `.kitchen.yml` start to fail

```
platforms:
- name: centos-6.4  # doesn't work w/ kitchen-vagrant 0.16.0
- name: centos-6.5  # doesn't work w/ kitchen-vagrant 0.16.0
- name: centos-6.6
- name: centos-7.0
- name: centos-7.1 # doesn't work w/ kitchen-vagrant 0.16.0
```

stopped working because that's not in their list of "current bento boxes" (which is already out of date, btw @fnichol). Using the atlas boxes seems better, as long as the box is there (and all the "old" ones still are)

```
platforms:
- name: chef/centos-6.4
- name: chef/centos-6.5
- name: chef/centos-6.6
- name: chef/centos-7.0
- name: chef/centos-7.1 # doesn't exist yet
```

Sorry about all the ranting

**TL;DR - is `chef/centos-7.1` going to be uploaded to atlas soon?**

**Comment by juliandunn** [comment]
This has been released to Atlas.

**Comment by jjasghar** [comment]
So yeah we need to make this more obvious, i ran into this problem myself. We should probably update .kitchen.yml.erb. I just opened: https://github.com/test-kitchen/test-kitchen/pull/730

**Comment by fnichol** [comment]
There's a few things going on here in parallel which eventually get us to a better place, so (for the moment) here is our landscape at the moment:

* Uploads to Atlas are done on an adhoc basis and usually aren't updated once initially uploaded.
* The same applies to the boxes hosted in S3 and linked from the [Bento README](https://github.com/chef/bento#current-baseboxes). Actually, most of the Atlas releases use this S3 bucket as the external hosting source.
* kitchen-vagrant currently assumes any platform name beginning with `centos`, `debian`, `fedora`, `freebsd`, `opensuse`, and `ubuntu` have a Bento box, hosted in the S3 bucket. It doesn't use Atlas by default.
* we can't make the `kitchen init` defaults have `"chef/"` prepended by default as that doesn't make sense if the user chooses to use the Docker, Openstack, EC2, or other Driver by default--it's more of a kitchen-vagrant default box name strategy which we can move to.

The reason kitchen-vagrant isn't yet using Atlas is that until we release updates to Atlas, the S3 artifacts stand the better chance of being updated. @b-dean, you're right that test-kitchen/kitchen-vagrant#151 was too aggressive and didn't allow for later releases of Bento boxes (which I'm sorry about).

The 0.18.0 release of kitchen-vagrant earlier this month tried to loosen the strategy of test-kitchen/kitchen-vagrant#151 in test-kitchen/kitchen-vagrant#166 which should pick up the `centos-7.1` box from the S3 bucket, as @juliandunn mentioned above.

There's some work ongoing in and around the Bento project to make versionable boxes suitable for uploading to Atlas in an automated way (in #358, #357, #360, #359, #361, #362, and #365 to date) which should help address the box freshness issue and publishing of new templates (hopefully) an easier process. Once the freshness of the Atlas box versions improves, we'll cut kitchen-vagrant over to using Atlas by default, so that a platform called `"centos-7.1"` would have a default box name of `"chef/centos-7.1"` automatically computed. This also makes `vagrant box outdated` and `vagrant box update` way more useful for users of Bento boxes.

Finally, we updated the default platform versions that `kitchen init` kicks out in test-kitchen/test-kitchen#711 (in master, pending a gem release), and the ChefDK generators in chef/chef-dk#382.

In short, yes we're still not at the ideal situation, but there's a good deal of forward movement to getting us there sooner rather than later. I hope this helps with background and context. Thanks! :heart:
